editor/plugins/gradient_editor_plugin.cpp

```diff
@@ -31,27 +31,583 @@
 #include "gradient_editor_plugin.h"
 
 #include "canvas_item_editor_plugin.h"
+#include "core/os/keyboard.h"
 #include "editor/editor_node.h"
 #include "editor/editor_scale.h"
 #include "editor/editor_settings.h"
+#include "editor/editor_spin_slider.h"
 #include "editor/editor_undo_redo_manager.h"
 #include "node_3d_editor_plugin.h"
+#include "scene/gui/flow_container.h"
+#include "scene/gui/separator.h"
 
-void GradientReverseButton::_notification(int p_what) {
+int GradientEditorRect::get_point_at(int x) {
+	int result = -1;
+	int total_w = get_gradient_rect_width();
+	float min_distance = 1e20;
```

Comment:
```suggestion
	float min_distance = handle_width * 0.8;
```
Allows `min` to be removed further down.